src/Adapter/GuzzleHttpAdapter.php

```diff
@@ -2,49 +2,65 @@
 
 namespace Elucidate\Adapter;
 
+use Elucidate\Exception\ElucidateRequestException;
+use Elucidate\Exception\ElucidateUncaughtException;
 use Elucidate\Model\RequestModel;
 use GuzzleHttp\Client;
+use GuzzleHttp\Exception\RequestException;
 use Psr\Http\Message\ResponseInterface;
 use Throwable;
+use Zend\Diactoros\Response\JsonResponse;
 
 class GuzzleHttpAdapter implements HttpAdapter
 {
     private $client;
+    private $suppressUncaughtExceptions;
 
     public function getBaseUri(): string
     {
         return $this->client->getConfig('base_uri');
     }
 
-    public function __construct(Client $client)
+    public function __construct(Client $client, bool $suppressUncaughtExceptions = true)
     {
         $this->client = $client;
+        $this->suppressUncaughtExceptions = $suppressUncaughtExceptions;
+    }
+
+    public function request(string $method, string $uri, array $options = [])
+    {
+        try {
+            return $this->client->request($method, $uri, $options);
+        } catch (RequestException $requestException) {
+            throw ElucidateRequestException::fromRequestException($requestException);
+        } catch (Throwable $e) {
+            if ($this->suppressUncaughtExceptions === false) {
+                throw new ElucidateUncaughtException("Uncaught exception", 500, $e);
```

Comment:
```throw $e;``` ?

There's not much else we can do with an uncaught exception except log it.

Reply:
Reasoning was because these exceptions are likely to be handled by Omeka, unfortunately it only seems to be able to be "scoped" in the handler by checking the controller. I really didn't want to tie it to that, so instead if it originated from Elucidate, its wrapped. See usage here that I want to replace: https://github.com/digirati-co-uk/omeka-elucidate-proxy-module/pull/1/commits/9bc978385efcc3b4c431735e5f02d0fc47ac65c2#diff-ea62a0c0e1b96c4f012299128648825eR39